src/test/java/com/nihongo/service/EmployeeServiceImplTest.java

```diff
@@ -0,0 +1,135 @@
+        employee.setUsername("Mohd Shoaib");
+        employee.setMail("mohd.shoaib@gmail.com");
+        employee.setDob("20-03-1999");
+        employee.setAddress("Bly");
+
+        when(employeeRepository.findById(1L)).thenReturn(Optional.of(employee));
+
+        List<Employee> employeeResponse =  employeeService.getEmployee(1L);
+
+        // Perform assertions
+        assertNotNull(employeeResponse);
+        assertEquals(1L, employeeResponse.get(0).getEmployeeId()); // Assuming you're comparing with a long value
+        assertEquals("Mohd Shoaib", employeeResponse.get(0).getUsername());
+        assertEquals("mohd.shoaib@gmail.com", employeeResponse.get(0).getMail());
+        assertEquals("20-03-1999", employeeResponse.get(0).getDob());
+        assertEquals("Bly", employeeResponse.get(0).getAddress());
+    }
+
+    @Test
+    void deleteEmployee() {
+        Employee employee = new Employee();
+        employee.setEmployeeId(1L);
+        employee.setUsername("Mohd Shoaib");
+        employee.setMail("mohd.shoaib@gmail.com");
+        employee.setDob("20-03-1999");
+        employee.setAddress("Bly");
+
+        employeeService.deleteEmployee(employee.getEmployeeId());
+
+        assertNotNull(employeeService);
+
+        // Verify that the deleteById method of the repository was called with the correct argument
+        verify(employeeRepository).deleteById(1L);
+    }
+
+    @Test
+    void saveEmployeeTest() {
+        EmployeePostRequest employeePostRequest = new EmployeePostRequest();
+        employeePostRequest.setUsername("Mohd Shoaib");
+        employeePostRequest.setEmail("mohd.shoaib@gmail.com");
+        employeePostRequest.setDob("20-03-1999");
+        employeePostRequest.setAddress("Bly");
+
+        Employee employee = new Employee();
+        employee.setEmployeeId(1L);
+        employee.setUsername("Mohd Shoaib");
+        employee.setMail("mohd.shoaib@gmail.com");
+        employee.setDob("20-03-1999");
+        employee.setAddress("Bly");
+
+        when(employeeRepository.save(any(Employee.class))).thenReturn(employee);
+
+        Employee employeeResponse = employeeService.saveEmployee(employeePostRequest);
+
+        assertNotNull(employeeResponse);
+        assertEquals(1L, employeeResponse.getEmployeeId()); // Assuming you're comparing with a long value
+        assertEquals("Mohd Shoaib", employeeResponse.getUsername());
+        assertEquals("mohd.shoaib@gmail.com", employeeResponse.getMail());
+        assertEquals("20-03-1999", employeeResponse.getDob());
+        assertEquals("Bly", employeeResponse.getAddress());
+     }
+
+    @Test
+    void updateEmployee() {
+        Employee employee = new Employee();
+        employee.setEmployeeId(1L);
+        employee.setUsername("Mohd Shoaib");
+        employee.setMail("mohd.shoaib@gmail.com");
+        employee.setDob("20-03-1999");
+        employee.setAddress("Bly");
+
+        when(employeeRepository.findById(any())).thenReturn(Optional.of(employee));
+
```

Comment:
Are we calling this findById method of employeeRepository class in updateEmployee method of service class?
I think no, then we do not need this mocking here.

Reply:
done